yt_dlp/extractor/tencent.py

```diff
@@ -97,21 +128,29 @@ def _extract_video_native_subtitles(self, api_response, subtitles_format):
 
         return subtitles
 
-    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
-        formats, subtitles = [], {}
-        for video_format, subtitle_format, video_quality in (
-                # '': 480p, 'shd': 720p, 'fhd': 1080p
-                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
-            api_response = self._get_video_api_response(
-                url, video_id, series_id, subtitle_format, video_format, video_quality)
+    def _extract_all_video_quality(self, api_response):
+        quality = []
+        formats_response = api_response['fl']['fi']
+        for f in formats_response:
+            quality.append(f.get('name'))
+        return quality
 
-            if api_response.get('em') != 0 and api_response.get('exem') != 0:
-                if '您所在区域暂无此内容版权' in api_response.get('msg'):
-                    self.raise_geo_restricted()
-                raise ExtractorError(f'Tencent said: {api_response.get("msg")}')
+    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
+        first_api_response = self._get_video_api_response(url, video_id, series_id, 'vtt', 'hls', 'hd')
+        self._check_api_response(first_api_response)
+        api_responses = [first_api_response]
+        quality = self._extract_all_video_quality(first_api_response)
+        for q in quality:
+            if q != 'sd' and q != 'hd':
+                api_response = self._get_video_api_response(
+                    url, video_id, series_id, 'vtt', 'hls', q)
```

Comment:
Could you try changing the first request from 'hd' to 'ld'? And try to reuse the response? I'd like to keep the number of requests at the minimum

I think this can be easier to read
```python
qualities = self._extract_all_video_qualities(first_api_response)
for quality in qualities:
      if quality not in ('sd',  'hd'):
           api_response = self._get_video_api_response(...)
```

Reply:
Now the number of API requests has been kept to a minimum, but I did not test WeTV, I will test it later.

Guest: only requested 3 times
```
python .\yt_dlp\__main__.py https://v.qq.com/x/cover/lqoxvyv6a23jo9o/b00440jumct.html --list-formats
[vqq:video] Extracting URL: https://v.qq.com/x/cover/lqoxvyv6a23jo9o/b00440jumct.html 
[vqq:video] b00440jumct: Downloading webpage
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[info] Available formats for b00440jumct: 
ID EXT RESOLUTION FPS │  TBR PROTO │ VCODEC   VBR ACODEC  ABR MORE INFO
────────────────────────────────────────────────────────────────────────
0  mp4 864x360     24 │ 456k m3u8  │ unknown 409k unknown 47k 标清 (480P)
1  mp4 864x360     24 │ 456k m3u8  │ unknown 409k unknown 47k 标清 (480P)
2  mp4 864x360     24 │ 456k m3u8  │ unknown 409k unknown 47k 标清 (480P)
3  mp4 864x360     24 │ 456k m3u8  │ unknown 409k unknown 47k 标清 (480P)
4  mp4 1280x534    24 │ 733k m3u8  │ unknown 671k unknown 63k 准高清 (720P)
5  mp4 1280x534    24 │ 733k m3u8  │ unknown 671k unknown 63k 准高清 (720P)
6  mp4 1280x534    24 │ 733k m3u8  │ unknown 671k unknown 63k 准高清 (720P)
7  mp4 1280x534    24 │ 733k m3u8  │ unknown 671k unknown 63k 准高清 (720P)
```

VIP Login user:
```
python .\yt_dlp\__main__.py https://v.qq.com/x/cover/lqoxvyv6a23jo9o/b00440jumct.html --list-formats --cookies-from-browser=chrome
[Cookies] Extracting cookies from chrome 
[Cookies] Extracted 1785 cookies from chrome
[vqq:video] Extracting URL: https://v.qq.com/x/cover/lqoxvyv6a23jo9o/b00440jumct.html 
[vqq:video] b00440jumct: Downloading webpage
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading webpage 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[vqq:video] b00440jumct: Downloading m3u8 information 
[info] Available formats for b00440jumct: 
ID EXT RESOLUTION FPS HDR   │   TBR PROTO │ VCODEC    VBR ACODEC   ABR MORE INFO
────────────────────────────────────────────────────────────────────────────────────
0  mp4 864x360     24       │  456k m3u8  │ unknown  409k unknown  47k 标清 (480P)
1  mp4 864x360     24       │  456k m3u8  │ unknown  409k unknown  47k 标清 (480P)
2  mp4 864x360     24       │  456k m3u8  │ unknown  409k unknown  47k 标清 (480P)
3  mp4 864x360     24       │  456k m3u8  │ unknown  409k unknown  47k 标清 (480P)
4  mp4 1280x534    24       │  733k m3u8  │ unknown  671k unknown  63k 准高清 (720P)
5  mp4 1280x534    24       │  733k m3u8  │ unknown  671k unknown  63k 准高清 (720P)
6  mp4 1280x534    24       │  733k m3u8  │ unknown  671k unknown  63k 准高清 (720P)
7  mp4 1280x534    24       │  733k m3u8  │ unknown  671k unknown  63k 准高清 (720P)
8  mp4 1920x800    24       │ 1263k m3u8  │ unknown 1169k unknown  93k 高清SDR (1080P)
9  mp4 1920x800    24       │ 1263k m3u8  │ unknown 1169k unknown  93k 高清SDR (1080P)
10 mp4 1920x800    24       │ 1263k m3u8  │ unknown 1169k unknown  93k 高清SDR (1080P)
11 mp4 1920x800    24       │ 1263k m3u8  │ unknown 1169k unknown  93k 高清SDR (1080P)
12 mp4 1920x800    24 hdr10 │ 1515k m3u8  │ unknown 1421k unknown  93k 臻彩视听 (臻彩) HDR
13 mp4 1920x800    24 hdr10 │ 1515k m3u8  │ unknown 1421k unknown  93k 臻彩视听 (臻彩) HDR
14 mp4 1920x800    24 hdr10 │ 1515k m3u8  │ unknown 1421k unknown  93k 臻彩视听 (臻彩) HDR
15 mp4 1920x800    24 hdr10 │ 1515k m3u8  │ unknown 1421k unknown  93k 臻彩视听 (臻彩) HDR
16 mp4 3840x1600   24       │ 2483k m3u8  │ unknown 2358k unknown 125k 超高清SDR (4K)
17 mp4 3840x1600   24       │ 2483k m3u8  │ unknown 2358k unknown 125k 超高清SDR (4K)
18 mp4 3840x1600   24       │ 2483k m3u8  │ unknown 2358k unknown 125k 超高清SDR (4K)
19 mp4 3840x1600   24       │ 2483k m3u8  │ unknown 2358k unknown 125k 超高清SDR (4K)
```

Reply:
The number of API requests for WeTV is now the same as the latest release.